### PR TITLE
task: add nem transfer and importance_transfer transactions

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

**Review by rg911 — APPROVED** [review]

File: nem/importance_transfer/importance_transfer.cats [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import "transaction.cats"
+
+# enumeration of importance transfer modes
+enum ImportanceTransferMode : uint32
+	# activates an importance transfer
+	link = 0x01
```
Comment (gimre-xymcity) [inline nem/importance_transfer/importance_transfer.cats:6]: should we maintain activate / deactivate naming from nis1?

File: nem/transfer/transfer.cats [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import "mosaic.cats"
+import "transaction.cats"
+
+# binary layout for a message
+struct Message
+	# reserved padding
+	reserved = make_reserved(uint32, 1)
```
Comment (gimre-xymcity) [inline nem/transfer/transfer.cats:7]: `, 0)`?

Comment (gimre-xymcity) [reply nem/transfer/transfer.cats:7]: actually what is this field?

File: nem/transfer/transfer.cats [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+# binary layout for a transfer transaction (V2)
+struct TransferTransaction2
+	version = make_const(uint8, 2)
+
+	inline TransferTransaction
+
+	# number of attached mosaics
+	mosaicsCount = uint8
+
```
Comment (gimre-xymcity) [inline nem/transfer/transfer.cats:45]: do we require comment _somewhere_ how amount is tied to mosaics attached 😬

(especially that you can have completely different transfers ending up with same amount transfered 😬)

File: nem/transfer/transfer.cats (hunk 2 above)
Comment (Jaguar0625) [reply nem/transfer/transfer.cats:7]: ah, that's message type (1 == plain). i guess i should add enum for that?

File: nem/transfer/transfer.cats (hunk 3 above)
Comment (segfaultxavi) [reply nem/transfer/transfer.cats:45]: We're [discussing over Discord](https://discord.com/channels/856325968096133191/864789538224341042/882627075641593876) the possibility that schema comments become the only source of truth for any schema-related documentation.

So YES, add as many comments as you can to the schemas so we have more meaningful docs in the end.

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**